Doc/library/statistics.rst

```diff
@@ -1054,6 +1069,46 @@ The final prediction goes to the largest posterior. This is known as the
   'female'
 
 
+Kernel density estimation
+*************************
+
+It is possible to estimate a continuous probability density function
+from a fixed number of discrete samples.
+
+The basic idea is to smooth the data using a kernel function
+such as a normal distribution, triangular distribution, or uniform
+distribution.  The degree of smoothing is controlled by a single
+parameter, ``h``, representing the variance of the kernel function.
+
+.. testcode::
+
+   import math
+
+   def kde_normal(sample, h):
+       "Create a continous probability density function from a sample."
+       # Smooth the sample with a normal distribution of variance h.
+       kernel_h = NormalDist(0.0, math.sqrt(h)).pdf
+       n = len(sample)
+       def pdf(x):
+           return sum(kernel_h(x - x_i) for x_i in sample) / n
+       return pdf
+
+`Wikipedia has an example
+<https://en.wikipedia.org/wiki/Kernel_density_estimation#Example>`_
+where we can use the ``kde_normal()`` recipe to generate and plot
+a probability density function estimated from a small sample:
+
+.. doctest::
```

Comment:
This block doesn't seem to test anything, perhaps we could change to ``.. code:: pycon``?

Reply:
FWIW, it does actually execute the statements and makes sure that they don't fail.

Reply:
When the 3.12 docs become unblocked, I'll backport this.

Reply:
Sounds good -- for reference I believe @Yhg1s is keeping the branch locked until 3.12.0 release, so the branch would be unlocked at some point in October.

> As a reminder, until the final release of 3.12.0, the 3.12 branch is set up so that the Release Manager (me) has to merge the changes.

from https://discuss.python.org/t/31137

Reply:
Backports to 3.12 are still happening. You should just need to assign the backport PR to Thomas so that he can merge it @rhettinger